scripts/pythondistdeps.py

```diff
@@ -334,83 +373,71 @@ def normalize_name(name):
                     spec = ('==', dist.py_version)
                     if spec not in py_deps[name]:
                         py_deps[name].append(spec)
-                deps = dist.requires()
+                deps = dist.requirements
                 if args.recommends:
-                    depsextras = dist.requires(extras=dist.extras)
-                    if not args.requires:
-                        for dep in reversed(depsextras):
-                            if dep in deps:
-                                depsextras.remove(dep)
-                    deps = depsextras
+                    deps = dist.extra_requirements
                 # console_scripts/gui_scripts entry points need pkg_resources from setuptools
-                if ((dist.get_entry_map('console_scripts') or
-                    dist.get_entry_map('gui_scripts')) and
+                if (dist.entry_points and
                     (lower.endswith('.egg') or
                      lower.endswith('.egg-info'))):
-                    # stick them first so any more specific requirement overrides it
-                    deps.insert(0, Requirement.parse('setuptools'))
+                    groups = set([ep.group for ep in dist.entry_points])
```

Comment:
```suggestion
                    groups = {ep.group for ep in dist.entry_points}
```

Unless we want to support Python 2.6 which I'd rather not (and most likely we already don't).